src/181810.c

```diff
@@ -442,9 +524,75 @@ s32 is_actortype_hpbar_visible(s32 actorType) {
     return ((get_global_byte(idx + 365) | battleStatus->tattleFlags[idx]) >> (actorType - (idx * 8))) & 1;
 }
 
-INCLUDE_ASM(s32, "181810", save_tattle_flags);
+void save_tattle_flags(s32 actorType) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    u8* list;
+    s32 gb;
+    s32 i = 0;
+    s32 j;
+    s32 k;
+
+    while (TRUE) {
+        list = ActorTypesLists[i];
+        if (list == NULL) {
+            break;
+        }
+
+        for (j = 0; list[j] != 0xFF; j++) {
+            if (actorType == list[j]) {
+                for (k = 0; list[k] != 0xFF; k++) {
+                    actorType = list[k];
+
+                    gb = get_global_byte((actorType / 8) + 0x16D);
+                    gb |= 1 << (actorType % 8);
+                    set_global_byte((actorType / 8) + 0x16D, gb);
+                    battleStatus->tattleFlags[actorType / 8] |= gb;
+                }
+                return;
+            }
+        }
+        i++;
+    }
+
+    gb = get_global_byte((actorType / 8) + 0x16D);
+    gb |= 1 << (actorType % 8);
+    set_global_byte((actorType / 8) + 0x16D, gb);
+    battleStatus->tattleFlags[actorType / 8] |= gb;
+}
+
+void load_tattle_flags(s32 actorType) {
+    BattleStatus* battleStatus = &gBattleStatus;
+    u8* list;
+    s32 gb;
+    s32 i = 0;
+    s32 j;
+    s32 k;
+
+    while (TRUE) {
+        list = ActorTypesLists[i];
+        if (list == NULL) {
+            break;
+        }
+
+        for (j = 0; list[j] != 0xFF; j++) {
+            if (actorType == list[j]) {
+                for (k = 0; list[k] != 0xFF; k++) {
+                    actorType = list[k];
+
+                    gb = get_global_byte((actorType / 8) + 0x16D);
```

Comment:
Can you add a note that GameByte(0x16D) is GB_Tattles_00? Alternatively, some macro for `(actorType / 8) + 0x16D` might be good given its done a few times